index.html

```diff
@@ -0,0 +1,173 @@
+<!DOCTYPE html>
+<html lang="ja">
+
+<head>
+    <meta charset="UTF-8">
+    <title>JavaScript Lesson4</title>
+</head>
+
+<body>
+    <div id="quiz-container" class="quiz-container">
+        <h2 id="quiz-title">ようこそ</h2>
+        <h3 id="quiz-category"></h3>
+        <h3 id="quiz-difficulty"></h3>
+        <div id="question" class="question">以下のボタンをクリック</div>
+        <ul id="answers"></ul>
+        <button id="start-button">開始</button>
+    </div>
+    <style>
+        ul {
+            list-style-type: none;
+            margin-block-start: 0;
+            margin-block-end: 0;
+            padding-inline-start: 0;
+        }
+        
+        .question {
+            margin-bottom: 2rem;
+            padding: .5rem 0;
+            border-top: 1px solid #000;
+            border-bottom: 1px solid #000;
+        }
+    </style>
+    <script>
+        const apiUrl = 'https://opentdb.com/api.php?amount=10';
+
+        const gameState = {
+            quizzes: [],
+            currentIndex: 0,
+            score: 0
+        };
+
+        const quizContainer = document.getElementById('quiz-container');
+        const quizTitle = document.getElementById('quiz-title');
+        const quizCategory = document.getElementById('quiz-category');
+        const quizDifficulty = document.getElementById('quiz-difficulty');
+        const quizContent = document.getElementById('question')
+        const answersContainer = document.getElementById('answers');
+        const startButton = document.getElementById('start-button');
+
+        startButton.addEventListener('click', (event) => {
+            getQuizData();
+        });
+
+        const getQuizData = () => {
+            quizTitle.textContent = '取得中です';
+            quizContent.textContent = '少々お待ちください';
+            startButton.hidden = true;
+
+            fetch(apiUrl)
+                .then((response) => {
+                    return response.json();
+                })
+                .then((data) => {
+                    gameState.quizzes = data.results;
+                    gameState.currentIndex = 0;
+                    gameState.score = 0;
+
+                    setNextQuiz();
+                });
+        };
```

Comment:
`async` `await` を使うとイマドキの記述になりますし、他の言語との差異も少ないのでおススメです。

```suggestion
        const getQuizData = async () => {
            quizTitle.textContent = '取得中です';
            quizContent.textContent = '少々お待ちください';
            startButton.hidden = true;

            const response = await fetch(apiUrl);
            const data = await response.json();
            gameState.quizzes = data.results;
            gameState.currentIndex = 0;
            gameState.score = 0;
            setNextQuiz();
        };
```